pkg/controllers/jobset_controller.go

```diff
@@ -355,22 +355,32 @@ func (r *JobSetReconciler) resumeJobSetIfNecessary(ctx context.Context, js *jobs
 				return err
 			}
 		}
+		// If using in order startup policy, return early and wait for the replicated job to be ready.
 		if inOrderStartupPolicy(startupPolicy) {
+			// Add a condition to the JobSet indicating the in order startup policy is executing.
 			return r.ensureCondition(ctx, ensureConditionOpts{
 				jobset:           js,
 				eventType:        corev1.EventTypeNormal,
 				forceFalseUpdate: true,
-				condition:        generateStartupPolicyCondition(metav1.ConditionFalse),
+				condition:        inOrderStartupPolicyExecutingCondition(),
 			})
 		}
 	}
+	// At this point all replicated jobs have had their jobs resumed.
+	// If using an in order startup policy. add a condition to the JobSet indicating the
+	// in order startup policy has completed.
 	if inOrderStartupPolicy(startupPolicy) {
-		return r.ensureCondition(ctx, ensureConditionOpts{
+		if err := r.ensureCondition(ctx, ensureConditionOpts{
```

Comment:
This is changing the code.

We want to return here if startup policy is done.

Reply:
Yeah sorry I forgot to put a note here explaining this change. My thought was this was a bug that we are returning early here, because otherwise we never set the condition below that indicates the JobSet has been resumed; we only add the condition that startup policy is done. Wdyt?

Reply:
I did that by design.

I return the jobset to trigger a reconcile. I only want to resume the jobset once all the jobs have resumed.

Reply:
Why I return is that I don't want to go to the next job in the loop until the previous job is started.

I think this could potentially start the next job once the previously is resumed.

Reply:
>Why I return is that I don't want to go to the next job in the loop until the previous job is started.

This is outside the loop though. Inside the loop we still do the early return on line 361, to wait for the replicatedJob we just resumed to be ready before moving on to the next.

Reply:
Ha. Yea I guess that is so..

Reply:
imo I would really try to keep bug fixes separate from refactor PRs. It makes cherry-picks and such much more difficult..

/hold

Reply:
Could you move this fix into a separate PR?